lib/components/sync-status/get-note-titles.ts

```diff
@@ -1,24 +1,25 @@
-import { compact } from 'lodash';
 import noteTitleAndPreview from '../../utils/note-utils';
 
-const getNoteTitles = (ids, notes, limit = Infinity) => {
-  const matchedNotes = ids.map((id, i) => {
-    if (i >= limit) {
-      return;
-    }
+import * as T from '../../types';
 
-    const note = notes.find(thisNote => thisNote.id === id);
-
-    if (!note) {
-      // eslint-disable-next-line no-console
-      console.log(`Could not find note with id '${id}'`);
-      return null;
-    }
+type NoteTitle = {
+  id: String;
+  title: String;
+};
 
-    return { id, title: noteTitleAndPreview(note).title };
-  });
+const getNoteTitles = (
+  ids: String[] = [],
+  notes: T.NoteEntity[] = [],
+  limit: number = Infinity
+) => {
+  const matchedNotes = ids.reduce((acc: NoteTitle[], id: String) => {
+    const note = notes.find((thisNote: T.NoteEntity) => thisNote.id === id);
```

Comment:
on the other side we can safely leave the annotations out of these two lines for `id` and `thisNote` because they follow from the types of `ids` and `notes`. if I leave out those annotations and hover over the identifiers my IDE shows the right types.

```ts
const matchedNotes = ids.reduce((acc: NoteTitle[], id) => {
  const note = notes.find(thisNote => thisNote.id === id);
  return note
    ? [ ...acc, { id, title: noteTitleAndPreview(note).title } ]
    : acc;
}, []);
```

Reply:
Fixed in: 95ea31d